### ISSUE TITLE
Image NoiseSampler

### ISSUE THREAD

**Issue by dfsek** [PolyhedralDev/Terra#67]
An implementation of NoiseSampler that uses an image with a configurable color channel to pull noise values from.

**Comment by dfsek** [comment]
this will also go along with a reimplementation of noise configs to make them more modular.

The reimplementation will use a nested approach for things like normalizers, and will allow definition of "sampler type" at the root, rather than noise type.
Each sampler type will have a separate sub-section. Mock config:
```yml
noise:
  sampler: NORMALIZER
  type: LINEAR
  linear:
    min: 0
    max: 1
  input:
    sampler: NOISE
    type: Cellular
    cellular:
      return: NoiseLookup
      lookup:
        sampler: NOISE
        type: OpenSimplex2
        fractal:
          type: FBm
          octaves: 2
        frequency: 0.01
    frequency: 0.005
```

This also gives us a chance to rename some of the not ideally named things from FastNoise (looking at you, cellular distance function types)

**Comment by dfsek** [comment]
Also this should for the most part maintain compatibility with old noise configs. (We can have `sampler` default to `NOISE` for backwards compat)
The only thing i see breaking are configs with normalizers, and that's an easy fix, besides, I doubt normalizers are widely known/used.

**Comment by dfsek** [comment]
Implemented in e306a0fd09e39f1ea17457dc0ff5252b097e3856
